### ISSUE TITLE
[jit][tracing] Preserve module type names during tracing

### ISSUE THREAD

**Issue by jerryzh168** [pytorch/pytorch#33913]
**What**: Preserve module type names during tracing
**Why**: Folding batch norm into conv is blocked for traced models currently. Right now we use module names to do ConvBatchNorm folding, so we need tracing to preserve module name in order to get automatic convbn folding in traced models
cc @suo @jamesr66a
